pycbc/inference/likelihood.py

```diff
@@ -0,0 +1,198 @@
+        A generator class that creates waveforms. This must have a generate
+        function which takes a set of parameter values as arguments, and a
+        detectors attribute which is a dictionary of detectors keyed by their
+        names.
+    data : dict
+        A dictionary of data, in which the keys are the detector names and the
+        values are the data (assumed to be unwhitened). The list of keys must
+        match the waveform generator's detectors keys.
+    f_lower : float
+        The starting frequency to use for computing inner products.
+    psds : {None, dict}
+        A dictionary of FrequencySeries keyed by the detector names. The
+        dictionary must have a psd for each detector specified in the data
+        dictionary. If provided, the inner products in each detector will be
+        weighted by 1/psd of that detector.
+    f_upper : {None, float}
+        The ending frequency to use for computing inner products. If not
+        provided, the minimum of the largest frequency stored in the data
+        and a given waveform will be used.
+    norm : {None, float or array}
+        An extra normalization weight to apply to the inner products. Can be
+        either a float or an array. If None, 4*data.values()[0].delta_f will be
+        used.
+    """
+
+    def __init__(self, waveform_generator, data, f_lower, psds=None,
+            f_upper=None, norm=None):
+        self._waveform_generator = waveform_generator
+        self._data = data
+        # check that the data and waveform generator have the same detectors
+        if sorted(waveform_generator.detectors.keys()) != \
+                sorted(self._data.keys()):
+            raise ValueError("waveform generator's detectors (%s) " %(
+                ','.join(sorted(waveform_generator.detector_names))) +
+                "does not match data (%s)" %(
+                ','.join(sorted(self._data.keys()))))
+        # check that the data sets all have the same lengths
+        dlens = numpy.array([len(d) for d in data.values()])
+        if not all(dlens == dlens[0]):
+            raise ValueError("all data must be of the same length")
+        N = dlens[0]
+        # we'll use the first data set for setting values
+        d = data.values()[0]
+        # figure out the kmin, kmax to use
+        kmin, kmax = filter.get_cutoff_indices(f_lower, f_upper, d.delta_f,
+            (N-1)*2)
+        self._kmin = kmin
+        self._kmax = kmax
+        if norm is None:
+            norm = 4*d.delta_f
+        # we'll store the weight to apply to the inner product
+        if psds is None:
+            w = norm*Array(numpy.ones(N))
+            self._weight = {det: w for det in data} 
+        else:
+            # temporarily suppress numpy divide by 0 warning
+            numpy.seterr(divide='ignore')
+            self._weight = {det: norm/psds[det] for det in data}
+            numpy.seterr(divide='warn')
+        # compute <d, d>
+        self._dd = {det:
```

Comment:
Okay this is pycbc 2.7 as well. And fixing it to python 2.6.6 syntax has things in ``pycbc.waveform.generator`` break. Did we move to 2.7?